rust/agama-lib/src/scripts/model.rs

```diff
@@ -454,4 +459,53 @@ mod test {
         _ = repo.clear();
         assert!(!script_path.exists());
     }
+
+    #[test]
+    async fn test_resolve_url_relative() {
+        let base_url = Uri::parse("http://example.lan/sles").unwrap().to_owned();
```

Comment:
```suggestion
        let base_url = Uri::parse("http://example.lan/sles.json").unwrap().to_owned();
```

to make it clear that it is not a directory